### PR TITLE
made changed to user schema

### PR DESCRIPTION
**Pull request by Noam-Moskowitz**

### CODE REVIEW

File: server/src/controllers/userController.ts [hunk 1]
```diff
@@ -1,120 +1,115 @@
...
+  static async addUser(req: Request, res: Response) {
     try {
-      const data = {
-        name: req.body.name,
-        email: req.body.email,
-      };
 
-      const { error, value } = UserSchemaValidation.validate(data);
+
+      const { error, value } = UserSchemaValidation.validate(req.body);
```
Comment (mike16345) [inline server/src/controllers/userController.ts]: put this in validation middleware.

File: server/src/services/userService.ts
Comment (mike16345) [inline server/src/services/userService.ts:1]: throw the errors

File: server/src/models/userModel.ts [hunk 2]
```diff
@@ -25,6 +56,12 @@ export const genToken = (id: string) => {
 };
 
 export const UserSchemaValidation = Joi.object({
-  name: Joi.string().min(2).max(25).required(),
-  email: Joi.string().min(2).max(30).required().email(),
+  firstName: Joi.string().min(2).max(25),
+  lastName: Joi.string().min(2).max(25),
+  email: Joi.string().min(5).max(30).email(),
+  password:Joi.string().optional(),
```
Comment (mike16345) [inline server/src/models/userModel.ts:62]: we wont be using passwords. maybe just in the beginning. either way we will have to encrypt the password in a seperate collection if we do.

File: server/src/interfaces/IUser.ts [hunk 3]
```diff
@@ -2,8 +2,13 @@ import { ObjectId } from "mongodb";
 
 export interface IUser {
   id: ObjectId;
```
Comment (mike16345) [inline server/src/interfaces/IUser.ts]: _id*

**Review by mike16345 — CHANGES_REQUESTED** [review]

File: server/src/models/userModel.ts (hunk 2 above)
Comment (Noam-Moskowitz) [reply server/src/models/userModel.ts:62]: yeah i know the password stuff is just there temporarily until we figure out the whole onBoarding signing in process

File: server/src/middleware/usersMiddleware.ts
Comment (mike16345) [inline server/src/middleware/usersMiddleware.ts:1]: bruv somehow keeps going back to the old formatting.

File: server/src/controllers/userController.ts [hunk 4]
```diff
@@ -35,12 +30,12 @@ export class UserController {
...
       if (!user) {
-        return res.status(404).send("User not found");
+        return res.status(StatusCode.NOT_FOUND).send("User not found");
       }
 
-      res.status(200).send(user);
+      res.status(StatusCode.OK).send(user);
     } catch (err: any) {
-      res.status(500).send({message:err});
+      res.status(StatusCode.NOT_FOUND).send({message:err});
```
Comment (mike16345) [inline server/src/controllers/userController.ts]: make sure when switching the status code to switch it to the same one. 

For example 500 is StatusCode.INTERNAL_SERVER_ERROR.

Now like i said it aint a big deal but if someone is expecting a code 500 and gets 404 then they will deal with it differently. So make sure you don't make changes like these without consulting the people who receive these responses.

**Pull request merged**